ui/pages/confirmations/hooks/useQueuedConfirmationEvents.ts

```diff
@@ -0,0 +1,35 @@
+import { useContext, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+  QueueType,
+} from '../../../../shared/constants/metametrics';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  getQueuedRequestCount,
+  pendingApprovalsSortedSelector,
+} from '../../../selectors';
+
+export const useQueuedConfirmationsEvent = (queueType: QueueType) => {
+  const pendingApprovals = useSelector(pendingApprovalsSortedSelector);
```

Comment:
This selector may be useful here: https://github.com/MetaMask/metamask-extension/blob/34543337da562145919c7f6c79d4c05e4683ab8b/ui/pages/confirmations/selectors/confirm.ts#L36

Reply:
Thanks Jyoti. However, this selector uses `pendingConfirmationsSortedSelector` which in turn filters the approval types. We want to show this banner for all confirmation types, including switch ethereum chain and others.